GVFS/GVFS.Service/GVFSMountProcess.Mac.cs

```diff
@@ -0,0 +1,47 @@
+﻿using GVFS.Common;
+using GVFS.Common.Tracing;
+using GVFS.Service.Handlers;
+using System;
+
+namespace GVFS.Service
+{
+    public class GVFSMountProcess : IDisposable
+    {
+        private const string ParamPrefix = "--";
+
+        private readonly ITracer tracer;
+
+        public GVFSMountProcess(ITracer tracer, int sessionId)
+        {
+            this.tracer = tracer;
+        }
+
+        public bool Mount(string repoRoot)
+        {
+            if (!this.CallGVFSMount(repoRoot))
+            {
+                this.tracer.RelatedError($"{nameof(this.Mount)}: Unable to start the GVFS process.");
+                return false;
+            }
+
+            string errorMessage;
+            if (!GVFSEnlistment.WaitUntilMounted(repoRoot, false, out errorMessage))
+            {
+                this.tracer.RelatedError(errorMessage);
+                return false;
+            }
+
+            return true;
+        }
+
+        public void Dispose()
+        {
+        }
+
+        private bool CallGVFSMount(string repoRoot)
+        {
+            InternalVerbParameters mountInternal = new InternalVerbParameters(startedByService: true);
+            throw new NotImplementedException();

```

Comment:
I'm having some trouble following what we expect to work and not work once this PR is merged in.

It looks like `Mount` is fully implemented, but `CallGVFSMount` is not.  What is the state of this class?

Reply:
What is implemented - Register, Un-Register and List Registered repositories.
What is Not implementd - Auto-mount, UT.